resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -373,20 +377,17 @@ private[spark] class Client(
       .map(CompressionUtils.createTarGzip(_))
   }
 
-  private def getDriverLauncherService(
-      k8ClientConfig: Config,
-      kubernetesMaster: String): KubernetesSparkRestApi = {
+  private def getDriverLauncherService(k8ClientConfig: Config): KubernetesSparkRestApi = {
```

Comment:
I think the library does that. 

See what I found 

The ConfigBuilder class is initialized with an instance of the config class with this [constructor](https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java#L131) which safely uses this [tryKubeConfig method ](https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java#L316)to read the kube config file.